control/control.go

```diff
@@ -0,0 +1,104 @@
+package control
+
+import (
+	"encoding/json"
+	"errors"
+	"net"
+	"net/http"
+	"os"
+	"sync"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+const (
+	SOCKET_TYPE = "unix"
+)
+
+// Server contains the state of the HTTP Server used by ContainerPilot's HTTP
+// transport control plane. Currently this is listening via a UNIX socket file.
+type HTTPServer struct {
+	mux        *http.ServeMux
+	addr       net.UnixAddr
+	listening  bool
+	lock       sync.RWMutex
+}
+
+// NewServer initializes a new control server for manipulating ContainerPilot's
+// runtime configuration.
+func NewHTTPServer(cfg *Config) (*HTTPServer, error) {
+	if cfg == nil {
+		err := errors.New("Control server not loaded due to missing config")
+		return nil, err
+	}
+
+	mux := http.NewServeMux()
+	addr := net.UnixAddr{
+		Name: cfg.SocketPath,
+		Net: SOCKET_TYPE,
+	}
+
+	return &HTTPServer{
+		mux: mux,
+		addr: addr,
+		listening: false,
+	}, nil
+}
+
+var listener net.Listener
+
+// Serve starts serving the control server
+func (s *HTTPServer) Serve() {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	// ref https://github.com/joyent/containerpilot/pull/165
+	if listener != nil {
+		return
+	}
+
+	s.mux.HandleFunc("/env", s.getEnvHandler)
+
+	ln, err := net.Listen(s.addr.Network(), s.addr.String())
+	if err != nil {
+		log.Fatalf("Error serving socket at %s: %v", s.addr.String(), err)
```

Comment:
_Super_ nitpicky but I'm trying to get all the error messages so they start with lower case unless starting with a proper name, so that when errors wrap each other they look nice. (Apparently this is a golang style thing.)